x/stakeibc/keeper/grpc_query_lsm_deposits.go

```diff
@@ -0,0 +1,72 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Stride-Labs/stride/v8/x/stakeibc/types"
+)
+
+func (k Keeper) LSMDeposit(c context.Context, req *types.QueryLSMDepositRequest) (*types.QueryLSMDepositResponse, error) {
+	if req == nil || req.GetChainId() == "" || req.GetDenom() == "" {
```

Comment:
Not a big deal at all, but we prefer accessing attributes of the protos directly (`req.ChainId`) instead of using the getters (`req.GetChainId()`) especially since you already check that `req` is not nil. 

We had a mix of both throughout our codebase, and decided we should start standardizing it, and the attribute approach is what we've seen throughout other projects and seems to be more of the standard in go

Reply:
On this I have to respectfully disagree.  I do agree we should be consistent but the Getter functions work in some cases where you can't do the direct property version.  In this exact situation it would be the same, but in the case of optional fields, the whole Xvalues stuff happening under the hood exposes the same GetValue() style getters.  If we want to be consistent we should do it that way.  Unrealted, it also makes it easier to mock that function in some places where you don't want to setup a ton of fake data for tests but you want to affect everything which gets returned for a certain case...

Reply:
Also might not affect us in this case but sometimes the value can be a null type whereas the Getters will coerce it to be the expected type -- in fact this is the source of the 0th enum problem, if there is no input a direct call to the property could get you a nil but the Getter will make sure to return a valid value of the expected type and for enums that default is the first legal value.  If we are doing checks expecting a certain value the Getters are slightly safer.

Reply:
I agree that if we did have optionals, this notation would make more sense, but I don't think we have any optional's in our codebase so that doesn't seem like a good case to optimize for. Also, I think the coercion does more harm than good. 

For instance, let's say we're trying to register an ICA channel and we need the chainId from the host zone struct for the channel name. At this point in our code base, if we have a nil host zone, we've got a big unexpected problem on our hands and the attribute approach would error as we'd prefer. But if we used a getter, this would pass an empty string for the chainID and we would register a channel with the incorrect name. This applies to most cases I think as well - we rarely pass around pointers, but when we do, if we somehow accidentally passed a nil, we should error right away, rather than masking the problem with a zero value.

But beyond that, imo the readability is much better when accessing with attributes, as (in go) Getters imply there's some calculation going on behind the scenes that's different than just accessing a field (contrary to a language like java).

But I'd also reiterate, that the attribute is common across every other codebase I've looked at (osmosis, cosmos-sdk, ibc-go, etc.) so I assume they're following a similar thinking.

Reply:
chatted offline, agreed on accessing using the attribute instead of the getters. marking this resolved